README.md

```diff
@@ -0,0 +1,175 @@
+
+# SSH Key Set Up
+
+Please see [this page](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account) for instructions. Feel free to message in #firmware on Slack with any questions!
+
+# Vagrant Environment Set Up for Windows Users
+
+### Overview
+
+Vagrant is an open-source software product for building and maintaining portable virtual software development environments. We will be using it to create an Ubuntu VM for VirtualBox.
+The repository we use is found [here](https://github.com/UWaterloo-Formula-Electric/vagrant).
+
+### Prerequisites
+
+- Install the latest version of [Vagrant](https://developer.hashicorp.com/vagrant/downloads)
+- Install the latest version of [VirtualBox](https://www.virtualbox.org/wiki/Downloads)
+- Install the matching [VirtualBox Extension Pack](https://www.virtualbox.org/wiki/Downloads) to get USB data
+  After installing the above, verify the Vagrant installation worked by opening a new command prompt or console, and checking that **vagrant** is available.
+
+```
+$ vagrant
+Usage: vagrant [options] <command> [<args>]
+
+    -v, --version                    Print the version and exit.
+    -h, --help                       Print this help.
+
+# ...
+```
+
+### Getting Started
+
+Make sure everything under **Prerequisites** has been installed, and that you have an SSH key added for your host computer to have GitHub access (check the link above).
+
+Then, run the following commands:
+
+```
+git clone git@github.com:UWaterloo-Formula-Electric/vagrant.git
+cd vagrant
+vagrant up
+vagrant reload
+```
+
+(may take a while to complete)
+The commands above clone the vagrant repo which you only need to do this one time in your set up. The next time, you can just call `vagrant up`. When its done, you will see a login window pop up. Do not maximize the window when on the login page (for some reason, this causes it to freeze). Click on the user "vagrant" and login with the password: **vagrant**.
+
+The **shared/** directory in the VM is shared between your host computer (your lapptop) and the virtual environment. The location in your host computer is the path to where you cloned the vagrant repo.
+
+To turn off the machine, run `vagrant halt` in the same terminal.
+
+### Checking ST Link
+
+The Vagrantfile is configured to automatically attach the ST Link to the VM, but we should make sure.
+
+First, check that your host computer detects the STM32 board when it's connected. If it's detected, with the board still connected, go into the VM, open a terminal, and run:`lsusb`
+
+You should see the following output:
+
+```
+Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
+Bus 002 Device 002: ID 0483:374b STMicroelectronics ST-LINK/V2.1 (Nucleo-F103RB)
+Bus 002 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
+```
+
+### Clone Firmware
+
+Clone the firmware repository into `shared/` directory. Navigate to where you cloned the **vagrant** repo.
```

Comment:
Might want to specify "In vagrant, clone the firmware repo....."

Reply:
Agreed, people get very confused on whats in their computer and whats run in the VM

Reply:
I would recommend doing it as "vagrant/shared/"